apps/api/src/modules/providers/sms-plivo/sms-plivo.service.ts

```diff
@@ -78,7 +78,14 @@ export class SmsPlivoService {
 
       return response;
     } catch (error) {
-      throw new Error(`Failed to send message: ${error.message}`);
+      if (error.apiID) {
+        // Log relevant parts of the error response
+        this.logger.error(`Error sent from provider: ${providerId}`, error.message);
+        throw error;
```

Comment:
I am assuming whereever we are throwing errors we have it being handled in parent classes?

Reply:
Could you elaborate a bit more and tell the following answers the query?

Each provider has a send methed, result of which is sent to `processNotificationQueue` present in `src/jobs/consumers/notifications/notification.consumer.ts`
This has a catch block for handling errors